### PR TITLE
Update documentation

### LINKED ISSUE #349: Editing a non-default local affects the default locale
While editing the Home page of my site in the "German" `locale` (where the default is "English (New Zealand)") I changed the title to "Home, lang:de", then clicked "Save and Publish".

This correctly updated `SiteTree_localised_live` to reflect the change `where Locale = 'de'`, however to my surprise then viewing the `SiteTree_live` table also revealed that `Title = 'Home, lang:de'`, which is somewhat both an erroneous state, and technically false.

This issue however did _not_ occur when editing the Content of the same page.

`tractorcow/silverstripe-fluent: 4.0.x-dev`
`commit ddf8c127328bb5d1f506a5e4b5b1bfdb99ebd1e9 (HEAD -> master, origin/master, origin/HEAD, composer/master)`

### PR DESCRIPTION
**Pull request by raissanorth**
The documentation changes alert users that editing non-default locales affects the SiteTree table and emphasise that localised content needs to be published in each locale one wants it to be visible. Fixes #349 and #350.

### CODE REVIEW

File: docs/en/configuration.md [hunk 1]
```diff
@@ -7,17 +7,30 @@ will interfere with locale bootstrapping in certain situations (such as `Securit
...
+**Note:** If using domains, you can additionally assign per-domain defaults as well.
+
  - `Domain`: Dropdown to assign this locale to a domain.
- - `Fallbacks`: Allows you to specify one or more fallback locales for this locale.
+ 
+Navigate to the `Fallbacks` tab, which allows you to specify one or more fallback locales for this locale.
+
+Once you add at least two locales to your site, you can begin localising your content. 
+
+***Important:** Pages in locales that fall back must be added and published in each locale you want them to be visible
```
Comment (robbieaverill) [inline docs/en/configuration.md]:
```diff
- ***Important:**
+ **Important:**
```

Comment (raissanorth) [reply docs/en/configuration.md]: As this point is rather important see #350, I intended to display this block in italics and bold. I am happy though to change it to something better suited - any ideas - to draw the necessary attention to it.

Comment (robbieaverill) [reply docs/en/configuration.md]: Ah, I see - in that case you need to close it correctly (notice that the bold carries through). Maybe use an underscore for italic instead

File: docs/en/configuration.md [hunk 2]
```diff
@@ -140,6 +153,9 @@ class FormPage extends Page
 }
 ```
 
+**Note:** Editing a non-default locale affects the `SiteTree(_live)` table. In contrast to SilverStripe 3, the
```
Comment (tractorcow) [inline docs/en/configuration.md]: It doesn't matter if the locale is default or not. In fluent 4 vs 3, the "default" locale matters a lot less than it once did as far as content authoring is going.

Now all it means is that the default will be the first locale a user sees on the frontend.

Comment (raissanorth) [reply docs/en/configuration.md]: Okay, I am going to change the phrasing slighlty. Let me know if that sounds better.

**Comment by tractorcow** [comment]
Thanks @raissanorth great job :D

**Pull request merged**